### ISSUE TITLE
Add `no-conflicting-checks` rule

### ISSUE THREAD

**Issue by lo1tuma** [marcalexiei/eslint-plugin-zod#366]
### 🧩 Description

Zod happily builds schemas whose checks contradict each other or don't even apply to the base type. Depending on the combo it silently matches nothing (`z.number().gt(10).lt(5)`, `z.string().uuid().email()`), silently no-ops (`z.number().check(z.minLength(1)`) accepts everything), or throws only at construction (`z.string().min(4).max(2)` → invalid regex `{4,2}`).

The `.check()` API is the sharp edge: it accepts any check regardless of the schema's type, so TypeScript can't help. `z.number().check(z.minLength(1))` compiles fine and silently no-ops, even though the chained `z.number().minLength()` is a compile error. A lint rule is the only way to catch these statically.

This rule flags `checks` that don't compose or don't apply. It's configurable by category (see options), each defaulting on. It covers chained methods, multi-argument `.check(...)`, and multiple `.check()` calls, or any weird combination of all of them, e.g. (`z.string().max(1).check(z.uuid(), z.email()).check(z.positive())`) in both plugins. It complements `no-duplicate-schema-methods`, which excludes `.check()` and doesn't inspect its arguments.

### 📦 Target plugin(s)

- [x] eslint-plugin-zod
- [x] eslint-plugin-zod-mini

### ⚙️ Options

```jsonc
{
    // provably unsatisfiable combinations — the schema can never match
    checkImpossibleCases?: boolean;    // default: true

    // technically valid but almost certainly a mistake: redundant / no-op combinations
    checkConfusingCases?: boolean;     // default: true

    // checks that don't apply to the schema's base type (only reachable via `.check()`,
    // since the chained equivalents are compile errors)
    checkInapplicableChecks?: boolean; // default: true
}
```

### 🧾 Example Configuration

```json
{
     "rules": {
        "zod/no-conflicting-checks": ["error", {
            "checkImpossibleCases": true,
            "checkConfusingCases": true,
            "checkInapplicableChecks": true
        }],
        "zod-mini/no-conflicting-checks": "error"
    }
}
```

### ✅ Valid Code Examples

```typescript
z.string().min(2).max(10);
z.array(z.string()).min(1).max(5);
z.number().gt(0).lt(100);
z.number().gte(5).lte(5);                     // exactly 5 — valid
z.number().int().gt(0).multipleOf(2);         // positive even integers
z.number().multipleOf(3).multipleOf(5);       // multiples of 15 — intentional
z.string().startsWith('a').startsWith('abc'); // 'a' is a prefix of 'abc'
z.string().min(5).includes('@');              // compatible
z.union([z.uuid(), z.email()]);               // either format — intentional
z.string().check(z.minLength(1), z.maxLength(10)); // applicable checks compose
z.number().check(z.gt(0), z.multipleOf(2));        // applicable checks compose
```

### ❌ Invalid Code Examples

```typescript
// ═══ Impossible: empty number range / contradictory sign ═══
z.number().gt(10).lt(5);
z.number().gte(5).lte(4);
z.number().gt(5).lt(5);
z.number().gt(5).lte(5);
z.number().gte(5).lt(5);
z.number().positive().negative();
z.number().positive().nonpositive();
z.number().negative().nonnegative();
z.number().positive().lte(0);
z.number().check(z.gt(10), z.lt(5));

// ═══ Impossible: length min > max / exact conflicts ═══
z.string().min(4).max(2);                        // throws at construction
z.string().length(3).min(4);                     // throws at construction
z.string().length(3).max(2);                     // throws at construction
z.array(z.string()).min(4).max(2);               // silent
z.string().length(2).length(3);
z.string().check(z.minLength(4), z.maxLength(2));
z.array(z.string()).check(z.minLength(4)).check(z.maxLength(2));

// ═══ Impossible: mutually-exclusive string formats (any two distinct formats) ═══
z.string().uuid().email();
z.string().email().url();
z.string().check(z.url(), z.email());
z.string().check(z.ipv4(), z.ipv6());
z.string().check(z.cidrv4(), z.cidrv6());
z.string().check(z.jwt(), z.uuid());
z.string().check(z.e164(), z.email());
z.string().check(z.mac(), z.uuid());
z.string().check(z.emoji(), z.ipv4());
z.string().check(z.iso.date(), z.iso.time());
z.string().check(z.iso.datetime(), z.iso.date());
z.string().check(z.cuid(), z.ulid());
z.string().check(z.nanoid(), z.xid());

// ═══ Impossible: format / content vs length ═══
z.string().uuid().max(1);                        // uuid is 36 chars
z.string().uuid().length(5);
z.string().check(z.ipv4(), z.maxLength(3));       // ipv4 needs >= 7 chars
z.string().check(z.jwt(), z.maxLength(2));
z.string().includes('xyz').max(2);               // needs >= 3 chars
z.string().startsWith('abcd').max(2);
z.string().startsWith('abcd').length(2);

// ═══ Impossible: incompatible prefixes / suffixes ═══
z.string().startsWith('foo').startsWith('bar');
z.string().endsWith('foo').endsWith('bar');

// ═══ Impossible: a literal contradicted by a check ═══
z.literal('foo').check(z.minLength(4));           // 'foo' is 3 chars
z.literal('foo').check(z.maxLength(2));
z.literal('foo').check(z.email());
z.literal('foo').check(z.startsWith('bar'));
z.literal(5).check(z.gt(10));
z.literal(5).check(z.negative());

// ═══ Confusing: technically valid, almost certainly a mistake ═══
z.string().lowercase().uppercase();              // only matches non-cased strings
z.number().gt(0).positive();                     // positive is just gt(0)
z.number().positive().nonnegative();             // nonnegative redundant
z.number().gt(0).gt(5);                           // gt(0) redundant
z.number().lt(10).lt(5);                          // lt(10) redundant
z.number().int().multipleOf(1);                  // multipleOf(1) redundant
z.number().multipleOf(2).multipleOf(4);          // multipleOf(2) redundant
z.string().max(10).max(5);                       // max(10) redundant
z.string().check(z.minLength(1), z.minLength(3)); // min(1) redundant, via .check()

// ═══ Confusing: a pointless check on a fixed / open type ═══
z.literal('foo').check(z.minLength(2));          // always true — pointless
z.literal(42).check(z.positive());               // always true — pointless
z.unknown().check(z.uuid());                     // constrains "unknown" to uuid — defeats the purpose
z.any().check(z.email());
z.never().check(z.uuid());                        // never already matches nothing

// ═══ Not applicable to the base type (only reachable via .check(); ═══
// ═══ the chained equivalents are compile errors) ═══
// number
z.number().check(z.minLength(1));                // length check — silently no-ops
z.number().check(z.startsWith('a'));             // string check — rejects all
z.number().check(z.uuid());                      // format — rejects all
// string
z.string().check(z.positive());
z.string().check(z.int());
z.string().check(z.multipleOf(2));
// array
z.array(z.string()).check(z.positive());
z.array(z.string()).check(z.email());
z.array(z.string()).check(z.startsWith('a'));
// boolean / date
z.boolean().check(z.minLength(1));
z.boolean().check(z.positive());
z.date().check(z.minLength(1));
z.date().check(z.email());
// bigint (has its own bigint bounds; number/length/format checks don't apply)
z.bigint().check(z.minLength(1));
z.bigint().check(z.uuid());
// number sub-types
z.int32().check(z.email());
z.uint32().check(z.startsWith('a'));
z.stringbool().check(z.minLength(1));
// object / strictObject / looseObject
z.object({}).check(z.positive());
z.strictObject({}).check(z.minLength(1));
z.looseObject({}).check(z.email());

// ═══ Several problems in one chain (each should be reported) ═══
z.string().max(1).check(z.uuid(), z.email()).check(z.positive());
// max(1) vs uuid length · uuid vs email · positive() on a string
z.number().gt(10).lt(5).check(z.minLength(2));
// empty range · length check on a number
z.string().uuid().email().check(z.positive());
// exclusive formats · number check on a string
z.array(z.string()).min(4).max(2).check(z.email());
// min > max · format check on an array
z.string().lowercase().uppercase().min(5).max(2);
// no-op case pair · min > max
z.literal('foo').check(z.email(), z.minLength(4));
// literal isn't an email · and is too short
```

### ✒️ Additional context

Type-inapplicable checks fail in two silent ways: a check that reads a missing property no-ops and accepts everything (`z.number().check(z.minLength(1))`), or it rejects everything (`z.string().check(z.positive())`), both give false confidence. This category only arises through `.check();` the chained methods are already type-safe.

Deliberately out of scope (too complex for a linter): anything needing number-theory or regex-intersection reasoning, e.g. `int()` combined with a range that contains no integer (`gt(0).lt(1)`), `multipleOf(k)` with a range containing no multiple, or general `regex()` compatibility.

**Comment by marcalexiei** [comment]
https://github.com/marcalexiei/eslint-zod/pull/369 should help implementing the rule. 
I'll do it once I have time.

**Comment by marcalexiei** [comment]
Done via #373.

Available in:
- [`eslint-plugin-zod@4.8.0`](https://github.com/marcalexiei/eslint-zod/releases/tag/eslint-plugin-zod%404.8.0)  
- [`eslint-plugin-zod-mini@1.5.0`](https://github.com/marcalexiei/eslint-zod/releases#release-eslint-plugin-zod-mini@1.5.0).
